File: app/src/main/java/com/neocaptainnemo/atms/model/ViewPort.java
package com.neocaptainnemo.atms.model;

import android.support.annotation.NonNull;

public class ViewPort {

    private double lngStart;
    private double latStart;
    private double lngEnd;
    private double latEnd;

    public ViewPort(double lngStart, double latStart, double lngEnd, double latEnd) {
        this.lngStart = lngStart;
        this.latStart = latStart;
        this.lngEnd = lngEnd;
        this.latEnd = latEnd;
    }

    public boolean isInside(@NonNull ViewPort another) {
        boolean res = (another.getLngStart() >= lngStart &&
                another.getLngStart() <= lngEnd &&
                another.getLatStart() >= latStart &&
                another.getLatStart() <= latEnd &&

                another.getLngEnd() >= lngStart &&
                another.getLngEnd() <= lngEnd &&
                another.getLatEnd() >= latStart &&
                another.getLatEnd() <= latEnd);
        return res;
    }

    public double getLngStart() {
        return lngStart;
    }

    public double getLatStart() {
        return latStart;
    }

    public double getLngEnd() {
        return lngEnd;
    }

    public double getLatEnd() {
        return latEnd;
    }
}

File: app/src/main/java/com/neocaptainnemo/atms/model/ViewPort.kt
package com.neocaptainnemo.atms.model

import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.SphericalUtil

data class ViewPort(val lngStart: Double, val latStart: Double, val lngEnd: Double, val latEnd: Double) {

    fun isInside(another: ViewPort): Boolean =
            another.lngStart >= lngStart &&
                    another.lngStart <= lngEnd &&
                    another.latStart >= latStart &&
                    another.latStart <= latEnd &&

                    another.lngEnd >= lngStart &&
                    another.lngEnd <= lngEnd &&
                    another.latEnd >= latStart &&
                    another.latEnd <= latEnd

    fun extended(): ViewPort {

        val southWest = SphericalUtil.computeOffset(LatLng(latStart, lngStart), 3000.0, 245.0)

        val startLng = southWest.longitude
        val startLat = southWest.latitude

        val northeast = SphericalUtil.computeOffset(LatLng(latEnd, lngEnd), 3000.0, 45.0)

        val endLng = northeast.longitude
        val endLat = northeast.latitude

        return ViewPort(startLng, startLat, endLng, endLat)
    }
}
